crates/optimism/payload/src/config.rs

```diff
@@ -100,6 +102,45 @@ struct OpDAConfigInner {
     max_da_block_size: AtomicU64,
 }
 
+/// Contains the Gas Limit configuration for the OP builder.
+///
+/// This type is shareable and can be used to update the Gas Limit configuration for the OP payload
+/// builder.
+#[derive(Debug, Clone, Default)]
+pub struct OpGasLimitConfig {
+    inner: Arc<OpGasLimitConfigInner>,
```

Comment:
since this just guards a single u64 I think we can replace this OpGasLimitConfigInner type with just AtomicU64